packages/eas-cli/src/fingerprint/utils.ts

```diff
@@ -0,0 +1,110 @@
+import { Platform, Workflow } from '@expo/eas-build-job';
+
+import { createFingerprintAsync } from './cli';
+import { Fingerprint } from './types';
+import { ExpoGraphqlClient } from '../commandUtils/context/contextUtils/createGraphqlClient';
+import { AppPlatform } from '../graphql/generated';
+import { FingerprintMutation } from '../graphql/mutations/FingerprintMutation';
+import Log from '../log';
+import { maybeUploadFingerprintAsync } from '../project/maybeUploadFingerprintAsync';
+import { resolveWorkflowPerPlatformAsync } from '../project/workflow';
+import { Client } from '../vcs/vcs';
+
+export async function getFingerprintInfoFromLocalProjectForPlatformsAsync(
+  graphqlClient: ExpoGraphqlClient,
+  projectDir: string,
+  projectId: string,
+  vcsClient: Client,
+  platforms: AppPlatform[]
+): Promise<Fingerprint> {
+  const workflows = await resolveWorkflowPerPlatformAsync(projectDir, vcsClient);
+  const optionsFromWorkflow = getFingerprintOptionsFromWorkflow(platforms, workflows);
+
+  const projectFingerprint = await createFingerprintAsync(projectDir, {
+    ...optionsFromWorkflow,
+    platforms: platforms.map(appPlatformToString),
+    debug: true,
+    env: undefined,
```

Comment:
an unrelated point here. some people use eas env var in their app config. passing undefined env may cause inconsistent fingerprint between local and server. i resolved that for eas update command from https://github.com/expo/eas-cli/pull/2867

Reply:
heads up @quinlanj